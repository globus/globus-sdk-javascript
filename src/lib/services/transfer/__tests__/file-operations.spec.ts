import serviceTestSuite from '../../../../__utils__/service-test-suite';
import { fileOperations } from '..';

/**
 * random endpoint-like identifier (uuid)
 */
const ENDPOINT = 'c591c905-2674-4227-9d31-1ff9485945a7';

serviceTestSuite('transfer', 'file-operations', (fetch) => {
  test('mkdir', async () => {
    await fileOperations.mkdir(ENDPOINT, {
      payload: {
        path: '/~/new-directory',
      },
    });
    expect(fetch).toHaveBeenCalled();
    expect(fetch).toHaveBeenCalledWith(
      `https://transfer.api.globusonline.org/v0.10/operation/endpoint/${ENDPOINT}/mkdir`,
      expect.objectContaining({
        body: expect.stringMatching('{"DATA_TYPE":"mkdir","path":"/~/new-directory"}') as unknown,
        headers: { 'Content-Type': 'application/json' },
        method: 'POST',
      }),
    );
  });

  test('rename', async () => {
    await fileOperations.rename(ENDPOINT, {
      payload: {
        old_path: '/~/old-directory',
        new_path: '/new-path',
      },
    });
    expect(fetch).toHaveBeenCalled();
    expect(fetch).toHaveBeenCalledWith(
      `https://transfer.api.globusonline.org/v0.10/operation/endpoint/${ENDPOINT}/rename`,
      expect.objectContaining({
        body: expect.stringMatching(
          '{"DATA_TYPE":"rename","old_path":"/~/old-directory","new_path":"/new-path"}',
        ) as unknown,
        headers: {
          'Content-Type': 'application/json',
        },
        method: 'POST',
      }),
    );
  });

  test('symlink', async () => {
    await fileOperations.symlink(ENDPOINT, {
      payload: {
        symlink_target: '/~/some/project/myfile.txt',
        path: '/~/quick_link.txt',
      },
    });
    expect(fetch).toHaveBeenCalled();
    expect(fetch).toHaveBeenCalledWith(
      `https://transfer.api.globusonline.org/v0.10/operation/endpoint/${ENDPOINT}/symlink`,
      expect.objectContaining({
        body: expect.stringMatching(
          '{"DATA_TYPE":"symlink","symlink_target":"/~/some/project/myfile.txt","path":"/~/quick_link.txt"}',
        ) as unknown,
        headers: {
          'Content-Type': 'application/json',
        },
        method: 'POST',
      }),
    );
  });

  test('ls', async () => {
    await fileOperations.ls(ENDPOINT, {
      query: {
        path: '/~/',
      },
    });
    expect(fetch).toHaveBeenCalled();
    expect(fetch).toHaveBeenCalledWith(
      `https://transfer.api.globusonline.org/v0.10/operation/endpoint/${ENDPOINT}/ls?path=%2F%7E%2F`,
      {
        headers: {},
      },
    );
  });
});
