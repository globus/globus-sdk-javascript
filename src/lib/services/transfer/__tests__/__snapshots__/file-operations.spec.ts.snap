// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`transfer.file-operations ls 1`] = `
{
  "headers": {
    "accept": "*/*",
    "accept-encoding": "gzip,deflate",
    "connection": "close",
    "host": "transfer.api.globusonline.org",
    "user-agent": "node-fetch/1.0 (+https://github.com/bitinn/node-fetch)",
  },
  "method": "GET",
  "url": "https://transfer.api.globusonline.org/v0.10/operation/endpoint/c591c905-2674-4227-9d31-1ff9485945a7/ls?path=%2F%7E%2F",
}
`;

exports[`transfer.file-operations mkdir 1`] = `
{
  "headers": {
    "accept": "*/*",
    "accept-encoding": "gzip,deflate",
    "connection": "close",
    "content-length": "47",
    "content-type": "application/json",
    "host": "transfer.api.globusonline.org",
    "user-agent": "node-fetch/1.0 (+https://github.com/bitinn/node-fetch)",
  },
  "json": {
    "DATA_TYPE": "mkdir",
    "path": "/~/new-directory",
  },
  "method": "POST",
  "url": "https://transfer.api.globusonline.org/v0.10/operation/endpoint/c591c905-2674-4227-9d31-1ff9485945a7/mkdir",
}
`;

exports[`transfer.file-operations rename 1`] = `
{
  "headers": {
    "accept": "*/*",
    "accept-encoding": "gzip,deflate",
    "connection": "close",
    "content-length": "75",
    "content-type": "application/json",
    "host": "transfer.api.globusonline.org",
    "user-agent": "node-fetch/1.0 (+https://github.com/bitinn/node-fetch)",
  },
  "method": "POST",
  "url": "https://transfer.api.globusonline.org/v0.10/operation/endpoint/c591c905-2674-4227-9d31-1ff9485945a7/rename",
}
`;

exports[`transfer.file-operations symlink 1`] = `
{
  "headers": {
    "accept": "*/*",
    "accept-encoding": "gzip,deflate",
    "connection": "close",
    "content-length": "96",
    "content-type": "application/json",
    "host": "transfer.api.globusonline.org",
    "user-agent": "node-fetch/1.0 (+https://github.com/bitinn/node-fetch)",
  },
  "method": "POST",
  "url": "https://transfer.api.globusonline.org/v0.10/operation/endpoint/c591c905-2674-4227-9d31-1ff9485945a7/symlink",
}
`;
